api/converter_utils.go

```diff
@@ -298,6 +308,94 @@ func txnRowToTransaction(row idb.TxnRow) (generated.Transaction, error) {
 	return txn, nil
 }
 
+func rowToBlock(blockHeader *sdk.BlockHeader) generated.Block {
+
+	rewards := generated.BlockRewards{
+		FeeSink:                 blockHeader.FeeSink.String(),
+		RewardsCalculationRound: uint64(blockHeader.RewardsRecalculationRound),
+		RewardsLevel:            blockHeader.RewardsLevel,
+		RewardsPool:             blockHeader.RewardsPool.String(),
+		RewardsRate:             blockHeader.RewardsRate,
+		RewardsResidue:          blockHeader.RewardsResidue,
+	}
+
+	upgradeState := generated.BlockUpgradeState{
+		CurrentProtocol:        string(blockHeader.CurrentProtocol),
+		NextProtocol:           strPtr(string(blockHeader.NextProtocol)),
+		NextProtocolApprovals:  uint64Ptr(blockHeader.NextProtocolApprovals),
+		NextProtocolSwitchOn:   uint64Ptr(uint64(blockHeader.NextProtocolSwitchOn)),
+		NextProtocolVoteBefore: uint64Ptr(uint64(blockHeader.NextProtocolVoteBefore)),
+	}
+
+	upgradeVote := generated.BlockUpgradeVote{
+		UpgradeApprove: boolPtr(blockHeader.UpgradeApprove),
+		UpgradeDelay:   uint64Ptr(uint64(blockHeader.UpgradeDelay)),
+		UpgradePropose: strPtr(string(blockHeader.UpgradePropose)),
+	}
+
+	var partUpdates *generated.ParticipationUpdates = &generated.ParticipationUpdates{}
+	if len(blockHeader.ExpiredParticipationAccounts) > 0 {
+		addrs := make([]string, len(blockHeader.ExpiredParticipationAccounts))
+		for i := 0; i < len(addrs); i++ {
+			addrs[i] = blockHeader.ExpiredParticipationAccounts[i].String()
+		}
+		partUpdates.ExpiredParticipationAccounts = strArrayPtr(addrs)
+	}
+	if len(blockHeader.AbsentParticipationAccounts) > 0 {
+		addrs := make([]string, len(blockHeader.AbsentParticipationAccounts))
+		for i := 0; i < len(addrs); i++ {
+			addrs[i] = blockHeader.AbsentParticipationAccounts[i].String()
+		}
+		partUpdates.AbsentParticipationAccounts = strArrayPtr(addrs)
+	}
+	if *partUpdates == (generated.ParticipationUpdates{}) {
+		partUpdates = nil
+	}
+
+	// order these so they're deterministic
+	orderedTrackingTypes := make([]sdk.StateProofType, len(blockHeader.StateProofTracking))
+	trackingArray := make([]generated.StateProofTracking, len(blockHeader.StateProofTracking))
+	elems := 0
+	for key := range blockHeader.StateProofTracking {
+		orderedTrackingTypes[elems] = key
+		elems++
+	}
+	sort.Slice(orderedTrackingTypes, func(i, j int) bool { return orderedTrackingTypes[i] < orderedTrackingTypes[j] })
```

Comment:
`slices.Sort` is a more convenient, and usually faster, way to sort.

Reply:
Addressed in https://github.com/algorand/indexer/pull/1638/commits/82e864bfb760ebec9056d80c83d63f7929f1eb65.

There are more instances of the `sort.Slice` function being used, but I don't want to pollute the diff. Can totally do this change (potentially in a separate PR) if you guys want.